Rules manager Panic caused by closing a channel more than once.

caused when sending a SIGHUP. and  SIGTERM right after that.
I looked briefly and seems like an easy one to solve.

```
panic: close of closed channel

goroutine 86182 [running]:
github.com/prometheus/prometheus/rules.(*Group).stop(...)
      ..../src/github.com/prometheus/prometheus/rules/manager.go:238
github.com/prometheus/prometheus/rules.(*Manager).Update.func1(0xc58eeacc01, 0xc42842c480, 0xc4200c5bc0, 0xc53ad459e0, 0xc44f874a20)
      ..../src/github.com/prometheus/prometheus/rules/manager.go:501 +0x80
created by github.com/prometheus/prometheus/rules.(*Manager).Update
      ..../src/github.com/prometheus/prometheus/rules/manager.go:499 +0x512
panic: close of closed channel

goroutine 86140 [running]:
github.com/prometheus/prometheus/rules.(*Group).stop(...)
      ..../src/github.com/prometheus/prometheus/rules/manager.go:238
github.com/prometheus/prometheus/rules.(*Manager).Update.func1(0xc4fab69b01, 0xc428228b40, 0xc4200c5bc0, 0xc53ad459e0, 0xc4e381def0)
      ..../src/github.com/prometheus/prometheus/rules/manager.go:501 +0x80
created by github.com/prometheus/prometheus/rules.(*Manager).Update
      ..../src/github.com/prometheus/prometheus/rules/manager.go:499 +0x512
```

--- a/cmd/prometheus/main.go
+++ b/cmd/prometheus/main.go
@@ -472,7 +472,9 @@ func main() {
 
 			},
 			func(err error) {
-				close(cancel)
+				// Wait for any in-progress reloads to complete to avoid
+				// reloading things after they have been shutdown.
+				cancel <- struct{}{}
 			},
 		)
 	}
@@ -502,6 +504,23 @@ func main() {
 				return nil
 			},
 			func(err error) {
+				close(cancel)
+			},
+		)
+	}
+	{
+		// Rule manager.
+		// TODO(krasi) refactor ruleManager.Run() to be blocking to avoid using an extra blocking channel.
+		cancel := make(chan struct{})
+		g.Add(
+			func() error {
+				<-reloadReady.C
+				ruleManager.Run()
+				<-cancel
+				return nil
+			},
+			func(err error) {
+				ruleManager.Stop()
 				close(cancel)
 			},
 		)
@@ -547,27 +566,7 @@ func main() {
 				return nil
 			},
 			func(err error) {
-				// Keep this interrupt before the ruleManager.Stop().
-				// Shutting down the query engine before the rule manager will cause pending queries
-				// to be canceled and ensures a quick shutdown of the rule manager.
 				cancelWeb()
-			},
-		)
-	}
-	{
-		// Rule manager.
-
-		// TODO(krasi) refactor ruleManager.Run() to be blocking to avoid using an extra blocking channel.
-		cancel := make(chan struct{})
-		g.Add(
-			func() error {
-				ruleManager.Run()
-				<-cancel
-				return nil
-			},
-			func(err error) {
-				ruleManager.Stop()
-				close(cancel)
 			},
 		)
 	}
@@ -626,6 +625,7 @@ func reloadConfig(filename string, logger log.Logger, rls ...func(*config.Config
 	if failed {
 		return fmt.Errorf("one or more errors occurred while applying the new configuration (--config.file=%s)", filename)
 	}
+	level.Info(logger).Log("msg", "Completed loading of configuration file", "filename", filename)
 	return nil
 }
 

